src/Confluent.Kafka/Admin/CreatePartitionsException.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+
+namespace Confluent.Kafka.Admin
+{
+    /// <summary>
+    ///     Represents an error that occured during a create partitions request.
+    /// </summary>
+    public class CreatePartitionsException : Exception
+    {
+        /// <summary>
+        ///     Initialize a new instance of CreatePartitionsException.
+        /// </summary>
+        /// <param name="results">
+        ///     The result corresponding to all topics in the request 
+        ///     (whether or not they were in error). At least one of these
+        ///     results will be in error.
+        /// </param>
+        public CreatePartitionsException(List<CreatePartitionsResult> results)
+            : base(
+                "An error occurred creating partitions for topics: [" +
+                String.Join(", ", results.Where(r => r.Error.IsError).Select(r => r.Topic)) +
+                "]: [" + String.Join(", ", results.Where(r => r.Error.IsError).Select(r => r.Error)) +
+                "].")
+        {
+            Results = results;
+        }
+        
+        /// <summary>
+        ///     The result corresponding to all topics in the request 
+        ///     (whether or not they were in error). At least one of these
+        ///     results will be in error.
+        /// </summary>
+        public List<CreatePartitionsResult> Results { get; }
```

Comment:
ditto, list. i did think about this a lot.

Reply:
So from my understanding you can pass any enumerable to CreatePartitions(), such as an unordered Dictionary.
With a List return type the ordering has no value and the application must scan the result list to find a desired entry - it seems to me that a result dictionary would be much more useful for the user.

Reply:
You can't pass a dictionary into the method as I have it defined - it's not compatible with: `IEnumerable<TopicSpecification>`. If you had a dictionary, it'd be a `IEnumerable<KeyValuePair<string, TopicSpecification>>`. If you do have a dictionary, you can convert it trivially like so: `specs.Select(s => s.Value)`. On the result side: `result.Single(r => r.Topic == "mytopic")`. Using a dictionary just forces you to go via `KeyValuePair<string, CreatePartitionsExceptionResult>` if you want to iterate through it. so, `foreach (var r in result) { r.Value.Error }`. The extra level of indirection doesn't read very well. Also, the only thing you want to do with the result is act on errors, and you'd typically want to do this via iteration, so that's why i write the interface that way. Also, using a dictionary here would be inconsistent with the rest of the library (all other methods that could take a dictionary take a flat enumerable).